dispatcher/README.md

```diff
@@ -0,0 +1,629 @@
+# CK Tile Dispatcher
+
+A unified kernel dispatch system for AMD GPUs with C++ and Python frontends.
+
+**Validated Platform:** AMD Instinct MI300 series (gfx942)
+
+---
+
+## Table of Contents
+
+1. [Quick Start](#quick-start)
+2. [Prerequisites](#prerequisites)
+3. [Step-by-Step Build Guide](#step-by-step-build-guide)
+4. [Running Examples](#running-examples)
+5. [External Integration](#external-integration)
+6. [Core Concepts](#core-concepts)
+7. [Troubleshooting](#troubleshooting)
+8. [File Structure](#file-structure)
+
+---
+
+## Quick Start
+
+**Complete setup from scratch (5 minutes):**
+
+```bash
+# From the composable_kernel root directory
+cd dispatcher
+
+# Step 1: Create build directory
+mkdir -p build && cd build
+
+# Step 2: Configure CMake
+cmake .. \
+  -DCMAKE_PREFIX_PATH=/opt/rocm \
+  -DCMAKE_CXX_COMPILER=/opt/rocm/bin/hipcc \
+  -DCMAKE_BUILD_TYPE=Release \
+  -DGPU_TARGETS="gfx942" \
+  -DBUILD_DISPATCHER_EXAMPLES=ON
+
+# Step 3: Generate kernels and build (CMake handles this automatically)
+make -j$(nproc)
+
+# Step 4: Run C++ examples
+./examples/gemm_01_basic
+./examples/conv_01_basic
+
+# Step 5: Run Python examples (from dispatcher directory)
+cd ..
+python3 examples/gemm/python/01_basic_gemm.py
+python3 examples/conv/python/01_basic_conv.py
+```
+
+---
+
+## Prerequisites
+
+### Required Software
+
+| Software | Minimum Version | Check Command |
+|----------|-----------------|---------------|
+| ROCm | 6.0+ | `rocminfo` |
+| CMake | 3.16+ | `cmake --version` |
+| Python | 3.8+ | `python3 --version` |
+| NumPy | Any | `pip show numpy` |
+| hipcc | (from ROCm) | `/opt/rocm/bin/hipcc --version` |
```

Comment:
I think some time around rocm 6.4 we were encouraged to switch from `hipcc` to clang bundled with the ROCm distribution